### ISSUE TITLE
PostgreSQLManagementClient.servers.create raises a misleading ValidationError about create_mode being None, when really it's a different property which is missing

### ISSUE THREAD

**Issue by sjelin** [Azure/azure-sdk-for-python#13126]
- **Package Name**: azure-mgmt-rdbms
- **Package Version**: 2.2.0
- **Operating System**: Ubuntu 18.04.3 LTS
- **Python Version**: 3.6.9

**Describe the bug**
Using [`PostgreSQLManagementClient.servers.create`](https://docs.microsoft.com/en-us/python/api/azure-mgmt-rdbms/azure.mgmt.rdbms.postgresql.operations.serversoperations?view=azure-python#create-resource-group-name--server-name--parameters--custom-headers-none--raw-false--polling-true----operation-config-) results in a `ValidationError` ~regardless of what you pass in, because [`ServerPropertiesForCreate.create_mode`](https://docs.microsoft.com/en-us/python/api/azure-mgmt-rdbms/azure.mgmt.rdbms.postgresql.models.serverpropertiesforcreate?view=azure-python#parameters) is required to be non-`None`, but is also [hardcoded to `None`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/rdbms/azure-mgmt-rdbms/azure/mgmt/rdbms/postgresql/models/_models.py#L1337)~ about `create_mode` being `None` when you pass in `create_mode` but miss some other parameter (e.g. the admin password).

This makes it very difficult to figure out what the correct parameters are.

**To Reproduce**
```
psql_client = ... # get PostgreSQLManagementClient instance
psql_client.servers.create('my_resource_group', 'my_server', {'properties': {'create_mode': 'whatever'}, 'location': 'eastus'})
```
results in
```
/opt/qumulo/env/202008140828/lib/python3.6/site-packages/msrest/serialization.py in validate(cls, data, name, **kwargs)
    686         required = kwargs.get('required', False)
    687         if required and data is None:
--> 688             raise ValidationError("required", name, True)
    689         elif data is None:
    690             return

ValidationError: Parameter 'ServerPropertiesForCreate.create_mode' can not be None.
```

EDIT: thanks to @changlong-liu for figuring out what this error message was trying to say and showing that this can be solved by specifying additional parameters:

```
psql_client.servers.create('my_resource_group', 'my_server', {'properties': {'create_mode': 'Default', "administrator_login": "xxx", "administrator_login_password": "xxx"}, 'location': 'eastus'})
```

**Expected behavior**
~If I pass in a non-`None` value for `create_mode`, I should not get a `ValidationError` about it being `None`.  If it must be hardcoded to `None`, it shouldn't be required.  If it must be required, it shouldn't be hardcoded to `None`.~

Edit:
The error message should say what is actually wrong with the parameters.  If the `create_mode` requires additional parameters, the error message should talk about those parameters.  If the `create_mode` is invalid, the error message should say so and list out what the valid options are.

**Additional context**

There is a work around, but it seems quite hack-y:
```
# EDIT: do not do this.  See changlong-liu's comment instead
server_props = ServerPropertiesForCreate() # no point in passing in `create_mode` here as it would be ignored
server_props.create_mode = 'Default'
psql_client = ... # get PostgreSQLManagementClient instance
psql_client.servers.create('my_resource_group', 'my_server', {'properties': server_props, 'location': 'eastus'})
```

~The documentation says that the field is a ["Constant filled by server"](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/rdbms/azure-mgmt-rdbms/azure/mgmt/rdbms/postgresql/models/_models.py#L1307), so it sounds like hardcoding it to `None` was correct, and the issue is just the validation.~

**Comment by rachel-msft** [comment]
Thanks for reaching out. We're investigating.

**Comment by changlong-liu** [comment]
Hi, 

for below error, it's not complaining about the ServerPropertiesForCreate.create_mode is None, but it's complaining that "Parameter 'ServerPropertiesForDefaultCreate.administrator_login' can not be None."
~~~
/opt/qumulo/env/202008140828/lib/python3.6/site-packages/msrest/serialization.py in validate(cls, data, name, **kwargs)
    686         required = kwargs.get('required', False)
    687         if required and data is None:
--> 688             raise ValidationError("required", name, True)
    689         elif data is None:
    690             return

ValidationError: Parameter 'ServerPropertiesForCreate.create_mode' can not be None.
~~~

Since you assign 'create_mode' as 'Default' in 'properties', the serializer parse 'properties' as an ServerPropertiesForDefaultCreate(https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/rdbms/azure-mgmt-rdbms/azure/mgmt/rdbms/postgresql/models/_models_py3.py#L1340).
Then it will require all required params in ServerPropertiesForDefaultCreate can be found.
An example for it:
~~~
psql_client.servers.create('my_resource_group', 'my_server', {'properties': {'create_mode': 'Default', "administrator_login": "xxx", "administrator_login_password": "xxx"}, 'location': 'eastus'})
~~~

**Comment by changlong-liu** [comment]
Hi @sjelin ,  

Would you help double check the error message?
In my side the error message is "Parameter 'ServerPropertiesForDefaultCreate.administrator_login' can not be None." for below payload:
~~~
psql_client.servers.create('my_resource_group', 'my_server', {'properties': {'create_mode': 'Default'}, 'location': 'eastus'})
~~~

If you assign any unexpected create_mode value, such as 'default', 'whaever', then it will report create_mode is None.
The available create_mode values can be found here: https://github.com/Azure/azure-sdk-for-python/blob/release/v3/sdk/rdbms/azure-mgmt-rdbms/azure/mgmt/rdbms/postgresql/models/_models_py3.py#L1326

**Comment by sjelin** [comment]
I think the error message `Parameter 'ServerPropertiesForDefaultCreate.administrator_login' can not be None` is perfect for `'properties': {'create_mode': 'Default'}`.

`create_mode is None` is not a good way to report that someone has passed in an invalid create mode.  I was very confused by that exact error message, and ended up wasting a lot of time thinking that I had the wrong method or keys or curly braces instead of realizing I just need to pass in a different value.  I only tried `'Default'` after hours of searching through documentation, giving up, reading code, and eventually finding `ServerPropertiesForDefaultCreate`.  If the error message had said `Invalid create_mode` instead, that would have saved me a lot of time.

If changing the error message is too difficult, then improving the documentation would also work.  [The current documentation for `.create`](https://docs.microsoft.com/en-us/python/api/azure-mgmt-rdbms/azure.mgmt.rdbms.postgresql.operations.serversoperations?view=azure-python#methods) leads you to [`ServerPropertiesForCreate`](https://docs.microsoft.com/en-us/python/api/azure-mgmt-rdbms/azure.mgmt.rdbms.postgresql.models.serverpropertiesforcreate?view=azure-python), which doesn't tell you what the valid create modes are, and searching through the codebase for the list is very difficult for non-contributors.  If that documentation had listed the valid create modes, I would have quickly figured out what I needed to change.
